crates/oxc_semantic/src/scope.rs

```diff
@@ -132,7 +132,7 @@ impl ScopeTree {
     pub fn set_parent_id(&mut self, scope_id: ScopeId, parent_id: Option<ScopeId>) {
         self.parent_ids[scope_id] = parent_id;
         if let Some(parent_id) = parent_id {
-            self.child_ids.entry(parent_id).or_default().push(scope_id);
+            self.child_ids[parent_id].push(scope_id);
```

Comment:
This could panic. `parent_id` should be inserted if it doesn't exist.

Reply:
I think the logic in `add_scope()` already takes care of this.
In particular, if we have a `ScopeId` for the parent here, it means that as some point we got it through a `let parent_id = add_scope(...);`. Within that execution, there has been a `self.child_ids.push(...)` call for the `parent_id` scope, which in turn ensures that this `self.child_ids[parent_id]` here always sees a populated entry.
Does this make sense?